### DISCUSSION TITLE
Stork 2.0.0 alpha releases are available

### DISCUSSION THREAD

**Discussion by jameslittle230** [jameslittle230/stork#318] [Announcements]
I've been working on Stork 2.0.0 over the past few months. It's a significant upgrade with a whole new search index format that will make searches faster and will make search indexes smaller.

I'm also taking the opportunity to build out a few features I've wanted Stork to have for a while. 

This means that Stork 1.x development will slow down, and most of my attention will be focused on finishing and releasing 2.0.0 I'll still make 1.x releases to fix bugs as needed - please let me know if there's something you're hoping I release in the 1.x track and I'll do my best to prioritize.

You can follow along with 2.0.0 development at [this branch](https://github.com/jameslittle230/stork/tree/v2). The latest alpha release is [2.0.0-alpha.3](https://github.com/jameslittle230/stork/releases/tag/v2.0.0-alpha.3).

Let me know if you have questions, comments, or significant feature requests!

**Comment by ndeville** [comment]
Just responded to an old thread - I'm really glad to see that making search indexes smaller is on the roadmap! 

Mine is 13Mb (so far), and I'm desperately looking for a solution. 

Thank you for all the hard work put into Stork!!

**Reply by whyboris** [reply]
Mine was about 8mb but was compressed down to 2mb with `brotli` compression 🥇 
Unfortunately it requires that the server serves the `.st` file with the proper headers (for the browser to know to decompress it first) 😅

**Reply by ndeville** [reply]
`brotli` compression cannot be setup on Github Pages, correct?

I have tried to use JS to remove Stork when accessing from mobile (as a workaround in the meantime), to no avail.. my JS skills are lacking. This is what I have used: 

``` html
<!-- for Stork seach engine -->
<!-- <script src="https://files.stork-search.net/stork.js"></script>
<script>
    stork.register("sitesearch", "{{ SITEURL }}/search-index.st")
</script> -->

<script>
    (function() {
    if( window.innerWidth > 600 ) {
        var stork_script = document.createElement('script');
        stork_script.type = 'text/javascript';
        stork_script.src = 'https://files.stork-search.net/stork.js';

        var stork_register = ('script');
        // stork_register.type = 'stork.register("sitesearch", "{{ SITEURL }}/search-index.st")';
        stork_register.innerHTML = 'stork.register("sitesearch", "{{ SITEURL }}/search-index.st")';

        var head = document.getElementsByTagName('head')[0];
        head.appendChild(stork_script);
        head.appendChild(stork_register);

        // stork_script.appendTo('body')
        // stork_register.appendTo('body')

    }
    })();
</script>
```

I have also found https://pagefind.app as a potential alternative - though wary of changing everything. 

I like Stork and would like to find a way to make it work.

**Comment by whyboris** [comment]
I'm unsure whether this was already considered, but it looks like we can unarchive (unzip?) _gzip_ compression with `WASM`, which means _Stork_ could compress the `.st` file before it's served 🤔 This means we don't have to worry about the static-file-server supporting the _gzip_ compression 🎉 

I think this may be all that's required: https://observablehq.com/@mourner/wasm-gunzip with `991 bytes` 🤯 

`brotli` compression would be better, but I'm unsure how big a `WASM` file we'd need to inflate the incoming `st` file 🤔

**Reply by jameslittle230** [reply]
This, along with sharded indexes, will primarily be how I accomplish making the search index smaller.

**Comment by jameslittle230** [comment]
Hi - I've released the first alpha prerelease of Stork 2.0.0 [here](https://github.com/jameslittle230/stork/releases/tag/v2.0.0-alpha.3). Feel free to give it a spin! I'm happy to answer any questions, since I recognize that it's very poorly documented.
